pkgs/games/sm64pc/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, fetchFromGitHub, requireFile, python3, pkg-config
+, audiofile, SDL2, hexdump, compileFlags ? [], region ? "us"
+, baseRom ? requireFile {
+    name = "baserom.${region}.z64";
+    message = ''
+      This nix expression requires that baserom.${region}.z64 is
+      already part of the store. To get this file you can dump your Super Mario 64 cartridge's contents
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+      Note that if you are not using a US baserom, you must overwrite the "region" attribute with either "eu" or "jp".
+    '';
+    sha256 = {
+      "us" = "17ce077343c6133f8c9f2d6d6d9a4ab62c8cd2aa57c40aea1f490b4c8bb21d91";
+      "eu" = "c792e5ebcba34c8d98c0c44cf29747c8ee67e7b907fcc77887f9ff2523f80572";
+      "jp" = "9cf7a80db321b07a8d461fe536c02c87b7412433953891cdec9191bfad2db317";
+    }.${region};
+  }
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sm64pc";
+  version = "unstable-2020-05-19";
+
+  src = fetchFromGitHub {
+    owner = "sm64pc";
+    repo = "sm64pc";
+    rev = "c18e70f44eaf628f5795002c29166bbfc4835238";
+    sha256 = "1vjk5g3jkrv21hwv13c7qf1disdv486sx6inn073ijx7g63hyzag";
+  };
+
+  nativeBuildInputs = [ python3 pkg-config ];
+  buildInputs = [ audiofile SDL2 hexdump ];
+
+  inherit baseRom;
+
+  makeFlags = [ "VERSION=${region}" ] ++ compileFlags;
+
+  preBuild = ''
+    substituteInPlace extract_assets.py --replace "/usr/bin/env python3" "${python3}/bin/python3"
```

Comment:
Oh, one more thing. You can use `patchShebangs extract_assets.py` or `patchShebangs .` for easier maintenance, but that's up to you. Every thing else looks good!

Reply:
Was not aware of that. Indeed a lot cleaner, I'll do that :+1: